.github/workflows/build.yml

```diff
@@ -48,20 +51,40 @@ jobs:
     - name: Bootstrap
       uses: ./.github/bootstrap_platform
 
-    - name: Configure CMake build
+     # We don't want to publish the test build, it gets too big.
+    - name: Build and Install (No tests)
       run: >
         ${{ matrix.config.preamble }}
 
         cmake -S . -B build -G Ninja
         --install-prefix ${{ github.workspace }}/dist
         --toolchain ${{ github.workspace }}/.conan/conan_paths.cmake
-        --preset test
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo
+
+        cmake --build build --config RelWithDebInfo
+
+        cmake --install build --config RelWithDebInfo
+
+    - name: Upload archive
+      uses: actions/upload-artifact@v3
+      with:
+        name: openassetio-${{ matrix.config.os }}
+        path: |
+          ${{ github.workspace }}/dist
 
-    - name: Build, install and test
+    # Reconfigure to add the test target. Ctest should rebuild using
+    # the cached build from prior.
+    - name: Test
       run: >
         ${{ matrix.config.preamble }}
 
-        ctest -VV --test-dir build --parallel 2
+        cmake -S . -B build -G Ninja
```

Comment:
[consider][minor] I'm pretty sure these switches don't need to be re-specified. I.e. you can run `cmake --preset test build`.  Not 100% on that when it comes to presets, needs checking.

Reply:
Would that also have the benefit that it makes it clearer we're intending to re-use the previous results?

Reply:
:+1: 
https://github.com/OpenAssetIO/OpenAssetIO/pull/1189/commits/6ecfa64019a63b8e8716f4b55d4ff9270b0759a7
https://github.com/OpenAssetIO/OpenAssetIO/pull/1189/commits/fe73992ebc3b01f58245d49ceb5bff6c3abf192b